src/components/views/admin/BuilderView.tsx

```diff
@@ -33,23 +32,20 @@ import { CandidateRateObj } from './types';
 
 const BuilderView: FC = () => {
   const { postAppData, patchAppData, appDataArray } = useAppDataContext();
-  const handlePatch = (
-    dataObj: AppData['id'],
-    newData: CandidateRateObj,
-  ): void => {
-    patchAppData({ id: dataObj, data: newData });
+  const handlePatch = (id: AppData['id'], newData: CandidateRateObj): void => {
+    patchAppData({ id, data: newData });
   };
   const handlePost = (newdata: CandidateRateObj): void => {
     postAppData({ data: newdata, type: APP_DATA_TYPES.CANDIDATE_RATE_INFO });
   };
-  const groupedData = appDataArray.groupBy((data) => data.type).toJS();
+  const groupedData = appDataArray.groupBy((data) => data.type).toJS() as {
+    [key: string]: AppData[];
+  };
   const [selectedType, setSelectedType] = useState('');
   const handleTypeChange = (value: string): void => {
     setSelectedType(value);
   };
-  const dataArray: any[] = Object.keys(groupedData)
-    .filter((key) => key === selectedType)
-    .map((key) => groupedData[key]);
+  const dataArray: AppData[] | undefined = groupedData[selectedType];
```

Comment:
If you type the selectedType state with the keys you expect it should be possible to never have `undefined`.